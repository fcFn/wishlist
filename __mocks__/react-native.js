const RN = require('react-native');

RN.AppState.removeEventListener = function() {};
RN.BackAndroid.removeEventListener = jest.fn();

module.exports = RN;
