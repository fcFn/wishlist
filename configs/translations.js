import I18n from 'react-native-i18n';

I18n.fallbacks = true;

I18n.translations = {
  en: {
    createNewListTitle: 'Create a new list',
    createNewListMsg: 'All unsaved changes will be lost. Proceed?',
    emptyList: 'The list is empty',
    newList: 'New list',
    saveList: 'Save list',
    loadList: 'Load list',
    genQR: 'Generate QR',
    readQR: 'Read QR',
    showHelpButton: 'Help',
    qrNote: 'Note: if you can\'t read the QR Code, your screen resolution and/or size may be too low for a list of this size',
    loadListTitle: 'Load a list',
    loadListMsg: 'All unsaved changes will be lost. Proceed?',
    editItem: 'Editing item',
    itemAlreadyExists: 'Item ${itemName} already exists',
    invalidQR: 'Invalid QR Code',
    qrListTooLarge: 'Your list is too large!',
    listSaved: 'List saved',
    listLoaded: 'List loaded',
    noListFound: 'No saved list found',
    readQRAlertTitle: 'Load from QR Code',
    readQRAlertMsg: 'Loading a list from QR Code will cause all unsaved changes to be lost. Proceed?',
    helpModalTitleFirstRun: 'Hello and welcome to Wishlist!',
    helpModalTitleNotFirstRun: 'Welcome back to Wishlist! Haven\'t seen you in a long time. Let me remind you how to use the app.',
    helpModalTitleOnDemand: 'Help',
    helpModalControlsAdd: 'To add an item, press the ⊕ button at the bottom of the list.',
    helpModalControlsRemove: 'To remove an item, swipe it to the left.',
    helpModalControlsEdit: 'To edit an item, press it and hold.',
    helpModalControlsMenu: 'To show additional options, press and hold the ⊕ button.',
    helpModalControlsMenuTitle: 'Additional options:',
    helpModalControlsMenuGenQR: 'Generate QR: generates a QR Code from the current list.',
    helpModalControlsMenuReadQR: 'Read QR: reads a QR Code generated in Wishlist and creates a list out of it.',
    helpModalControlsMenuLoad: 'Load list: loads a previously saved list.',
    helpModalControlsMenuSave: 'Save list: saves the current list, overwriting the previously saved list.',
    helpModalControlsMenuNew: 'New list: creates a new list.',
    helpModalControlsHelp: 'Help: shows this window.',
    helpModalCloseButton: 'Close',
  },
  ru: {
    createNewListTitle: 'Новый список',
    createNewListMsg: 'Несохранённые данные будут потеряны. Продолжить?',
    emptyList: 'Список пуст',
    newList: 'Новый список',
    saveList: 'Сохранить список',
    loadList: 'Загрузить список',
    genQR: 'Сгенерировать QR',
    readQR: 'Считать QR',
    showHelpButton: 'Помощь',
    qrNote: 'Если Вы не можете считать QR код, разрешение или размер вашего экрана не подходит для списка такого размера',
    loadListTitle: 'Загрузить список',
    loadListMsg: 'Несохранённые данные будут потеряны. Продолжить?',
    editItem: 'Редактирование пункта',
    itemAlreadyExists: 'Пункт ${itemName} уже в списке',
    invalidQR: 'Некорректный QR Code',
    qrListTooLarge: 'Список слишком большой!',
    listSaved: 'Список сохранён',
    listLoaded: 'Список загружен',
    noListFound: 'Сохранённый список не найден',
    readQRAlertTitle: 'Загрузить из QR кода',
    readQRAlertMsg: 'При загрузке списка из QR кода все несохранённые данные будут потеряны. Продолжить?',
    helpModalTitleFirstRun: 'Приветствую! Спасибо, что вы воспользовались приложением «Покупки»!',
    helpModalTitleNotFirstRun: 'Приветствую! Давно не виделись! Позвольте напомнить, как пользоваться этим приложением.',
    helpModalTitleOnDemand: 'Помощь',
    helpModalControlsRemove: 'Чтобы удалить пункт, проведите по нему пальцем влево.',
    helpModalControlsEdit: 'Чтобы отредактировать пункт, нажмите на него и удерживайте.',
    helpModalControlsAdd: 'Чтобы добавить пункт, нажмите кнопку «⊕», расположенную внизу списка.',
    helpModalControlsMenu: 'Для перехода к дополнительным опциям, нажмите и удерживайте кнопку «⊕».',
    helpModalControlsMenuTitle: 'Дополнительные опции:',
    helpModalControlsMenuGenQR: 'Сгенерировать QR: создаёт QR код из текущего списка.',
    helpModalControlsMenuReadQR: 'Считать QR: считывает QR код, сгенерированный в программе, и отображает закодированный в нём список.',
    helpModalControlsMenuLoad: 'Загрузить список: загружает сохранённый ранее список.',
    helpModalControlsMenuSave: 'Сохранить список: сохраняет текущий список, перезаписывая предыдущее сохранение.',
    helpModalControlsMenuNew: 'Новый список: создаёт новый список.',
    helpModalControlsHelp: 'Помощь: показать это окно.',
    helpModalCloseButton: 'Закрыть',
  },
};
