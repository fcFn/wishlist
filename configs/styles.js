import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#F5FCFF',
  },
  helpModalRoot: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  menuModalRoot: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
  },
  helpModalContent: {
    width: 340,
    marginTop: 7,
    backgroundColor: 'white',
    alignSelf: 'center',
  },
  helpModalContentText: {
    padding: 10,
    paddingTop: 5,
    lineHeight: 15,
    fontWeight: 'bold',
  },
  menuModalContent: {
    width: 300,
    marginTop: 10,
    backgroundColor: 'white',
    alignSelf: 'center',
  },
  item: {
    height: 40,
    borderWidth: 2,
    borderTopWidth: 0,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    paddingLeft: 10,
  },
  helpModalCloseButton: {
    marginBottom: 10,
    width: 150,
    height: 45,
    backgroundColor: 'lightblue',
    alignSelf: 'center',
    borderRadius: 4,
  },
  helpModalCloseButtonText: {
    alignSelf: 'center',
    marginTop: 12,
    fontWeight: 'bold',
  },
  itemText: {
    fontSize: 25,
  },
  itemDone: {
    backgroundColor: 'lightgreen',
  },
  itemTouched: {
    backgroundColor: 'lightblue',
  },
  itemRemove: {
    backgroundColor: 'red',
  },
  itemFlashy: {
    backgroundColor: 'green',
  },
  buttons: {
    flexDirection: 'row',
  },
  addButton: {
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
    borderWidth: 2,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    flex: 1,
    height: 45,
    backgroundColor: '#7de4ff',
  },
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderLeftWidth: 2,
    borderRightWidth: 2,
    borderTopWidth: 0,
    borderBottomWidth: 2,
    height: 45,
    backgroundColor: '#7de4ff',
  },
  newListButton: {
    borderRightWidth: 0,
  },
  addButtonText: {
    fontSize: 30,
    fontWeight: 'bold',
  },
  buttonText: {
    fontSize: 15,
    fontWeight: 'bold',
    paddingLeft: 5,
    paddingRight: 5,
  },
  newItemInput: {
    flex: 1,
    backgroundColor: 'white',
  },
  textInputStyleAddEditItem: {
    color: 'black',
    backgroundColor: 'white',
    height: 45,
  },
});

styles.itemColors = {
  inactive: 'white',
  checked: 'lightgreen',
  remove: 'red',
  active: 'lightblue',
  alreadyExists: 'green',
};
export default styles;
