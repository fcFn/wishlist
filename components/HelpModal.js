// This component is used to display the modal containing help information

import React, {Component} from 'react';
import {Text, Modal, View, TouchableNativeFeedback} from 'react-native';
import I18n from 'react-native-i18n';
import styles from '../configs/styles';

export default class HelpModal extends Component {

  render() {
    let title;
    switch (this.props.type) {
    case 'first':
      title = 'helpModalTitleFirstRun';
      break;
    case 'notFirst':
      title = 'helpModalTitleNotFirstRun';
      break;
    case 'onDemand':
      title = 'helpModalTitleOnDemand';
    }
    return <Modal
    animationType={'none'}
    transparent={true}
    visible={this.props.visible}
    onRequestClose={this.props.closeHelpModal}
  >
    <View style={styles.helpModalRoot}>
      <View style={styles.helpModalContent}>
        <Text style={styles.helpModalContentText}>
          {I18n.t(title) + '\n\n'}
          {I18n.t('helpModalControlsAdd') + '\n\n'}
          {I18n.t('helpModalControlsRemove') + '\n\n'}
          {I18n.t('helpModalControlsEdit') + '\n\n'}
          {I18n.t('helpModalControlsMenu') + '\n\n'}
          {I18n.t('helpModalControlsMenuTitle') + '\n\n'}
          {I18n.t('helpModalControlsMenuGenQR') + '\n\n'}
          {I18n.t('helpModalControlsMenuReadQR') + '\n\n'}
          {I18n.t('helpModalControlsMenuLoad') + '\n\n'}
          {I18n.t('helpModalControlsMenuSave') + '\n\n'}
          {I18n.t('helpModalControlsMenuNew') + '\n\n'}
          {I18n.t('helpModalControlsHelp')}
        </Text>
        <TouchableNativeFeedback onPress={this.props.closeHelpModal}>
          <View style={styles.helpModalCloseButton}>
            <Text style={styles.helpModalCloseButtonText}>
                {I18n.t('helpModalCloseButton')}
            </Text>
          </View>
        </TouchableNativeFeedback>
      </View>
    </View>
  </Modal>;
  }
}
