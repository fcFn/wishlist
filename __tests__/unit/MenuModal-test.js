import React from 'react';
import 'react-native';
import HelpModal from '../../components/MenuModal';
import '../../configs/translations';
import renderer from 'react-test-renderer';

describe('Menu modal window', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HelpModal />);
    expect(tree).toMatchSnapshot();
  });
});
