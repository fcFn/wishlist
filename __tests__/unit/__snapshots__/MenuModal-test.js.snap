// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu modal window renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={undefined}
  transparent={true}
  visible={true}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "marginTop": 10,
          "width": 300,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
